www/__tests__/notifScheduler.test.ts

```diff
@@ -0,0 +1,453 @@
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'schedule')
+      .mockImplementation((arg, callback) => callback(arg));
+    // call the function
+    await updateScheduledNotifs(reminderSchemes, isScheduling, setIsScheduling, scheduledPromise);
+
+    expect(setIsScheduling).toHaveBeenCalledWith(true);
+    expect(logDebug).toHaveBeenCalledWith('After cancelling, there are no scheduled notifications');
+    expect(logDebug).toHaveBeenCalledWith('After scheduling, there are no scheduled notifications');
+    expect(setIsScheduling).toHaveBeenCalledWith(false);
+  });
+
+  it('should resolve without scheduling if notifications are already scheduled', async () => {
+    // updateScheduleNotifs arguments
+    const reminderSchemes: any = exampleReminderSchemes;
+    let isScheduling: boolean = false;
+    const setIsScheduling: Function = jest.fn((val: boolean) => (isScheduling = val));
+    const scheduledPromise: Promise<any> = Promise.resolve();
+    // create the mock notifs from cordova plugin (must match the notifs that will generate from the reminder scheme above...
+    // in this case: exampleReminderSchemes.weekly, because getUser is mocked to return reminder_assignment: 'weekly')
+    const mockNotifs = [
+      { trigger: { at: DateTime.fromFormat('2023-11-14 21:00', 'yyyy-MM-dd HH:mm').toMillis() } },
+      { trigger: { at: DateTime.fromFormat('2023-11-15 21:00', 'yyyy-MM-dd HH:mm').toMillis() } },
+      { trigger: { at: DateTime.fromFormat('2023-11-17 21:00', 'yyyy-MM-dd HH:mm').toMillis() } },
+      { trigger: { at: DateTime.fromFormat('2023-11-19 21:00', 'yyyy-MM-dd HH:mm').toMillis() } },
+    ];
+
+    // mock the cordova plugin
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'getScheduled')
+      .mockImplementation((callback) => callback(mockNotifs));
+    // call the function
+    await updateScheduledNotifs(reminderSchemes, isScheduling, setIsScheduling, scheduledPromise);
+
+    expect(logDebug).toHaveBeenCalledWith('Already scheduled, not scheduling again');
+  });
+
+  it('should wait for the previous scheduling to finish if isScheduling is true', async () => {
+    // updateScheduleNotifs arguments
+    const reminderSchemes: any = exampleReminderSchemes;
+    let isScheduling: boolean = true;
+    const setIsScheduling: Function = jest.fn((val: boolean) => (isScheduling = val));
+    const scheduledPromise: Promise<any> = Promise.resolve();
+    // create an empty array of mock notifs from cordova plugin
+    const mockNotifs = [];
+
+    // mock the cordova plugin
+    jest
+      .spyOn(window['cordova'].plugins.notification.local, 'getScheduled')
+      .mockImplementation((callback) => callback(mockNotifs));
+    // call the function
+    await updateScheduledNotifs(reminderSchemes, isScheduling, setIsScheduling, scheduledPromise);
+
+    expect(logDebug).toHaveBeenCalledWith(
+      'ERROR: Already scheduling notifications, not scheduling again',
+    );
```

Comment:
In this instance, though, I think checking the `logDebug` does make sense because we expect `updateScheduledNotifs` to have quit early and not scheduled any notifications.

Reply:
I agree with:

> The logDebug statements are good for dev purposes but I don't think they should be the focus of our testing strategy.

In a future fix, I would suggest changing the underlying function to return a value that you can check, and then changing the test to check it